File: lib/shared_worker.dart
import 'dart:html' as html;

 final worker = html.SharedWorker('shared_worker.js');
// Create and manage the Shared Worker
void setupSharedWorker(Function(dynamic eventData) callback) {
 

  // Listen for messages from the worker
  worker.port!.onMessage.listen((event) {
    callback(event.data);
  });

  
}


void handleClickData(String q){
// Send a message to the worker
  worker.port!.postMessage(q);
}

File: web/shared_worker.js
// shared_worker.js
const connections = [];

onconnect = (event) => {
  const port = event.ports[0];
  connections.push(port);

  port.onmessage = (messageEvent) => {
    for (const conn of connections) {
      if (conn !== port) {
        conn.postMessage(messageEvent.data);
      }
    }
  };

  port.onclose = () => {
    const index = connections.indexOf(port);
    if (index !== -1) {
      connections.splice(index, 1);
    }
  };
};
